internal/lang/funcs/string.go

```diff
@@ -164,3 +166,129 @@ func Replace(str, substr, replace cty.Value) (cty.Value, error) {
 func StrContains(str, substr cty.Value) (cty.Value, error) {
 	return StrContainsFunc.Call([]cty.Value{str, substr})
 }
+
+// MakeTemplateStringFunc constructs a function that takes a string and
+// an arbitrary object of named values and attempts to render that string
+// as a template using HCL template syntax.
+func MakeTemplateStringFunc(content string, funcsCb func() map[string]function.Function) function.Function {
+
+	params := []function.Parameter{
+		{
+			Name:        "data",
+			Type:        cty.String,
+			AllowMarked: true,
+		},
+		{
+			Name:        "vars",
+			Type:        cty.DynamicPseudoType,
+			AllowMarked: true,
+		},
+	}
+	loadTmpl := func(content string, marks cty.ValueMarks) (hcl.Expression, error) {
+
+		// This condition checks if the provided string to be rendered as a template is marked as sensitive.
+		// If the string is marked as sensitive, it returns an error indicating that sensitive strings cannot be used as template strings.
+		if strings.Contains(marks.GoString(), "Sensitive") {
```

Comment:
yeah i'll update this.

Reply:
I dont think I need to explicitly add any check marking the resultant value as sensitive because, this is already implemented in opentofu itself. (screenshots attached)

<img width="1661" alt="Screenshot 2024-02-12 at 4 16 45 PM" src="https://github.com/opentofu/opentofu/assets/76054960/d5b5245f-81c2-4f74-ab9e-b1f85fe66e3e">

<img width="1732" alt="Screenshot 2024-02-12 at 4 17 31 PM" src="https://github.com/opentofu/opentofu/assets/76054960/fb335b59-2405-4a97-9294-1cf2c50a94f5">

Reply:
I'm not quite sure what you mean here. I'm suggesting two things here:

- checking the marks in a different way, instead you're pulling it out manually and checking against the string, instead of using `marks.Has`

- checking the marks in a different location (This could've happened one level higher) 

however we've now decided to propogate the sensitivity so this check is now redundant, right?

Reply:
yeah right.